lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -471,27 +493,37 @@ export default class BitcoinProcessor {
 
       const firstValidTransaction = await this.firstValidTransaction(exponentiallySpacedTransactions);
 
-      let revertToTransactionNumber: number;
-
       if (firstValidTransaction) {
-        // The number that represents the theoritical last possible transaction written on `firstValidTransaction.transactionTime`.
-        revertToTransactionNumber = TransactionNumber.construct(firstValidTransaction.transactionTime + 1, 0) - 1;
-      } else {
-        const lowestHeight = exponentiallySpacedTransactions[exponentiallySpacedTransactions.length - 1].transactionTime;
-        revertToTransactionNumber = TransactionNumber.construct(lowestHeight, 0);
-      }
+        // Revert all transactions in blocks from revertToBlockNumber and later. We make make this to be a group
+        // boundary to simplify resetting proof-of-fee state which is maintained per group.
+        let revertToBlockNumber = this.roundToGroupBoundary(firstValidTransaction.transactionTime);
 
-      console.debug(`Removing transactions since ${TransactionNumber.getBlockNumber(revertToTransactionNumber)}`);
-      await this.transactionStore.removeTransactionsLaterThan(revertToTransactionNumber);
+        // The number that represents the theoritical last possible transaction written with a block number
+        // less than revertToBlockNumber
+        const revertToTransactionNumber = TransactionNumber.construct(revertToBlockNumber + 1, 0) - 1;
 
-      // If we have reverted back to a valid transaction, rollback is complete.
-      if (firstValidTransaction) {
-        console.info(`reverted Transactions to block ${firstValidTransaction.transactionTime}`);
-        return firstValidTransaction.transactionTime;
+        console.debug(`Removing transactions since ${TransactionNumber.getBlockNumber(revertToTransactionNumber)}`);
+        await this.transactionStore.removeTransactionsLaterThan(revertToTransactionNumber);
+
+        // Reset transaction sampling
+        this.transactionSampler.clear();
+
+        // Revert the quantile calculator
+        await this.quantileCalculator.removeGroupsGreaterThanOrEqual(revertToBlockNumber + 1);
```

Comment:
For my understanding, any particular reason why have `removeGroupsGreaterThanOREQUAL()` and pass a `+1` instead of just `removeGroupsGreaterThan()` and not pass a `+1`?